### PR TITLE
Blueprint/api

### PR DESCRIPTION
**Pull request by johncookds**
Blueprint API and tests. Added dummy controller functions for route testing purposes

### CODE REVIEW

**Review by bonotake — COMMENTED** [review]
You can use `controllers.retrieve()` currently proposed in #18 instead of `get_relations()` and `get_events()`.

**Comment by johncookds** [comment]
Just updated to match those changes, tests should fail until #18 is merged

File: relations/routes/api.py [hunk 1]
```diff
@@ -21,3 +21,48 @@ def service_status() -> Response:
     response.status_code = status_code
     response.headers.extend(headers)
     return response
+
+@blueprint.route('/<string:arxiv_id_str>v<int:arxiv_ver>/relations', methods=['POST'])
+def create_new(arxiv_id_str: str, arxiv_ver: int) -> Response:
+    """create a new relation for an e-print"""
```
Comment (erickpeirson) [inline relations/routes/api.py]:
```suggestion
    """Create a new relation for an e-print."""
```

File: relations/routes/api.py [hunk 2]
```diff
@@ -21,3 +21,48 @@ def service_status() -> Response:
...
+    response_data, status_code, headers = controllers.create_new(arxiv_id_str, arxiv_ver, request.data)
+    response: Response = jsonify(response_data)
+    response.status_code = status_code
+    response.headers.extend(headers)
+    return response
+
+@blueprint.route('/<string:arxiv_id_str>v<int:arxiv_ver>/relations/<string:relation_id_str>', methods=['PUT'])
+def supercede(arxiv_id_str: str, arxiv_ver: int, relation_id_str: str) -> Response:
+    """create a new relation for an e-print which supersedes an existing relation"""
```
Comment (erickpeirson) [inline relations/routes/api.py]:
```suggestion
    """Create a new relation for an e-print which supersedes an existing relation."""
```

File: relations/routes/api.py [hunk 3]
```diff
@@ -21,3 +21,48 @@ def service_status() -> Response:
...
+    response_data, status_code, headers = controllers.supercede(arxiv_id_str, arxiv_ver, relation_id_str, request.data)
+    response: Response = jsonify(response_data)
+    response.status_code = status_code
+    response.headers.extend(headers)
+    return response
+
+@blueprint.route('/<string:arxiv_id_str>v<int:arxiv_ver>/relations/<string:relation_id_str>', methods=['DELETE'])
+def suppress(arxiv_id_str: str, arxiv_ver: int, relation_id_str: str) -> Response:
+    """create a new relation for an e-print which supresses an existing relation"""
```
Comment (erickpeirson) [inline relations/routes/api.py]:
```suggestion
    """Create a new relation for an e-print which supresses an existing relation."""
```

File: relations/routes/api.py [hunk 4]
```diff
@@ -21,3 +21,48 @@ def service_status() -> Response:
...
+    response_data, status_code, headers = controllers.suppress(arxiv_id_str, arxiv_ver, relation_id_str, request.data)
+    response: Response = jsonify(response_data)
+    response.status_code = status_code
+    response.headers.extend(headers)
+    return response
+
+@blueprint.route('/<string:arxiv_id_str>v<int:arxiv_ver>', methods=['GET'])
+def get_relations(arxiv_id_str: str, arxiv_ver: int) -> Response:
+    """get all active (not suppressed or superseded) relations for an e-print"""
```
Comment (erickpeirson) [inline relations/routes/api.py]:
```suggestion
    """Get all active (not suppressed or superseded) relations for an e-print."""
```

File: relations/routes/api.py [hunk 5]
```diff
@@ -21,3 +21,48 @@ def service_status() -> Response:
...
+    response_data, status_code, headers = controllers.retrieve(arxiv_id_str, arxiv_ver, active_only=True)
+    response: Response = jsonify(response_data)
+    response.status_code = status_code
+    response.headers.extend(headers)
+    return response
+
+@blueprint.route('/<string:arxiv_id_str>v<int:arxiv_ver>/log', methods=['GET'])
+def get_events(arxiv_id_str: str, arxiv_ver: int) -> Response:
+    """get the complete set of relation events (including suppressed and superseded)"""
```
Comment (erickpeirson) [inline relations/routes/api.py]:
```suggestion
    """Get the complete set of relation events (including suppressed and superseded)."""
```

File: relations/routes/tests/test_api.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    """Sample tests for external API routes."""
+
+    def setUp(self) -> None:
+        """Initialize the Flask application, and get a client for testing."""
+        os.environ['JWT_SECRET'] = 'foosecret'
+        self.app = create_api_app()
+        self.client = self.app.test_client()
+
+    @mock.patch(f'{api.__name__}.controllers.create_new')
+    def test_create_new(self, mock_create_new: Any) -> None:
```
Comment (erickpeirson) [inline relations/routes/tests/test_api.py:29]: Totally fine to annotate tests, but not necessarily required. When we do static checks e.g. on Travis we usually ``| grep -v "test.*"``

**Review by erickpeirson — APPROVED** [review]
Looks good! Minor suggestions on docstring formatting.

**Comment by erickpeirson** [comment]
@johncookds Looks like all systems go. Merge when ready! :-)

**Pull request merged**